packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -369,7 +369,7 @@ export const DataProviderMixin = (superClass) =>
       this._hasData = false;
       this.__updateVisibleRows();
 
-      if (!this._flatSize) {
+      if (!this._flatSize || !this.__virtualizer.size) {
```

Comment:
Can we actually replace `this._flatSize` with `this.__virtualizer.size` check here?

Reply:
I did consider it but I wonder if it's possible to end up in an inverse situation (where flat size is 0 but the virtualizer size is still above 0)...

Reply:
Removed `this._flatSize` after all. Couldn't come up with how the situation mentioned above would manifest.